ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/mainnet/MainnetTransactionValidator.java

```diff
@@ -37,7 +38,7 @@
  */
 public class MainnetTransactionValidator implements TransactionValidator {
 
-  public static final int NO_CHAIN_ID = -1;
+  public static final BigInteger NO_CHAIN_ID = BigInteger.valueOf(-1);
```

Comment:
Shouldn't be using a sentinel value here - I suspect this constant can just go away and `Optional.empty()` used in its place.

Reply:
done